src/content/en/2019/third-parties.md

```diff
@@ -169,12 +171,16 @@ The interest of site owners in understanding and analyzing user behavior is not
 
-## Conclusion
+### Security
 
-This chapter has shown that use of third-party content is rampant. This is entirely unsurprising - the entire basis of the web is to allow interconnectedness and linking.
+While the topic of security is covered more in-depth in a [separate chapter](./security#vulnerable-js-libraries), the security implications of introducing external dependencies to your site go hand-in-hand with privacy concerns. Allowing third parties to execute arbitrary JavaScript effectively provides them with complete control over your page. When a script can control the DOM and `window`, it can do everything. Even if code has no security concerns it can introduce a single point of failure [which has been recognized as a potential problem for some time now](https://www.stevesouders.com/blog/2010/06/01/frontend-spof/).
+
+[Self-hosting third-party content](https://csswizardry.com/2019/05/self-host-your-static-assets/) addresses some of the concerns mentioned here - and others. Additionally with browsers increasingly [partitioning HTTP caches](https://chromestatus.com/feature/5730772021411840) the benefits of loading directly from the third-party are increasingly questionable. Perhaps this is a better way to consume third-party content for many use cases, even if it makes measuring its impact more difficult.
+
+## Conclusion
 
-[Reuse in computer technologies](https://en.wikipedia.org/wiki/Code_reuse) is generally a good thing, though it does introduce dependencies which can impact performance (in positive and negative ways!) and has privacy implications, both of which we have touched on here. We have also not discussed the security implications of adding such dependencies to your site, though the security chapter has a [section on vulnerable JS libraries](./security#vulnerable-js-libraries) which is relevant and may be of interest to readers. Even if code has no security concerns it can introduce a single point of failure [which has been recognized as a potential problem for some time now](https://www.stevesouders.com/blog/2010/06/01/frontend-spof/).
+Third-party content is everywhere. This is hardly surprising; the entire basis of the web is to allow interconnectedness and linking. In this chapter we have examined third-party content in terms of assets hosted away from the main domain. If we had included self-hosted third-party content (e.g. common open source libraries hosted on the main domain), third-party usage would have be even larger!
```

Comment:
just edited a bit to be more consistent with the way I probably would've written it, no real substance here but lmk if anything here is more confusing :)